File: mobile/lib/api/recommendation_api.dart
import 'api_client.dart';

class RecommendationApi {
  static Future<Map<String, dynamic>> getAll() => ApiClient.get('/recommendations');
  static Future<Map<String, dynamic>> getToday() => ApiClient.get('/recommendations/today');
  static Future<Map<String, dynamic>> accept(int id) => ApiClient.post('/recommendations/$id/accept');
  static Future<Map<String, dynamic>> reject(int id) => ApiClient.post('/recommendations/$id/reject');
  static Future<Map<String, dynamic>> feedback(int id, String text) =>
      ApiClient.post('/recommendations/$id/feedback', body: {'feedback': text});
}

File: frontend/src/api/recommendationApi.js
import api from './axiosInstance';

export const recommendationApi = {
  getAll: () => api.get('/recommendations'),
  getToday: () => api.get('/recommendations/today'),
  accept: (id) => api.post(`/recommendations/${id}/accept`),
  reject: (id) => api.post(`/recommendations/${id}/reject`),
  feedback: (id, data) => api.post(`/recommendations/${id}/feedback`, data),
};
